sqltap/sqltap.py

```diff
@@ -296,6 +338,8 @@ def __init__(self, stats, report_file=None, report_dir=".",
 
         :param template_dir: folder of the template to generate the report.
         """
+        self.duration = ((stats[-1].end_time - stats[0].start_time)
+                         if stats else 0)
```

Comment:
If no queries are captured, then `stats` is falsy and the expression returns 0 from the else clause.

Where exactly would the profiling session store this overall duration and how would it be passed into `sqltap.report`? Note that your `generate_reports` example throws away the original collection and generates a fresh list. I wager that >90% of all SQLTap users are doing exactly this.